### PR TITLE
PHP: use php7_wrapper to reduce duplicated codes(final part)

### PR DESCRIPTION
**Pull request by thinkerou**
First part #7498 and second part #7532 , the PR includes:

- Add macro `PHP_GRPC_FREE_WRAPPED_FUNC_START` and `PHP_GRPC_FREE_WRAPPED_FUNC_END`

    - use it to unify `free_wrapped_grpc_xxx` function on php5 and php7 to one macro function

- Rename `wrapped_grpc_call_creds_from_obj` to `wrapped_grpc_call_credentials_from_obj`, etc, because macro `PHP_GRPC_FREE_WRAPPED_FUNC_START` on php7 need to use it

- Add macro `PHP_GRPC_HASH_FOREACH_xxx` about iterate hashtable

- Simplify `create_wrapped_grpc_xxx` function

- Add macro `php_grpc_zend_hash_find`

- Add macro `php_grpc_zend_hash_del`

@stanley-cheung Please review the PR which includes all change I know, especially I think the PR change about `hashtable` you should important review. Thanks.

### CODE REVIEW

File: src/php/ext/grpc/call.c [hunk 1]
```diff
@@ -185,96 +142,48 @@ zval *grpc_parse_metadata_array(grpc_metadata_array
...
-      metadata->metadata[metadata->count].key = key;
-      metadata->metadata[metadata->count].value = Z_STRVAL_P(*value);
-      metadata->metadata[metadata->count].value_length = Z_STRLEN_P(*value);
-      metadata->count += 1;
-    }
-  }
-#else
-  ZEND_HASH_FOREACH_STR_KEY_VAL(array_hash, key, inner_array) {
-    if (key == NULL) {
+  char *key1 = NULL;
```
Comment (thinkerou) [inline src/php/ext/grpc/call.c]: It is `key`, for solving macro conflict, so I use `key1`.

**Comment by stanley-cheung** [comment]
The last change looks good.

**Comment by thinkerou** [comment]
But `gcov` is wrong on `php7`.

**Comment by stanley-cheung** [comment]
I disabled `gcov` on `php7`. The red icon is for previous runs

**Comment by stanley-cheung** [comment]
@thinkerou Looks like both PRs are looking good with the tests now. I will probably merge these tomorrow. Anything else on this effort to reduce code duplication? If not, I expect to lock down the `v1.0.x` branch for PHP soon. Any new change should only go to the `master` branch. 

Again thanks for the contribution and great work here!

**Comment by thinkerou** [comment]
About reducing code duplication I think I have not commit temporary. But some areas still use `PHP_MAJOR_VERSION`, I think we think about how optimize it later and complete `TODO`. So you can lock down the `v1.0.x` branch. What about you?

**Comment by thinkerou** [comment]
Oh, wait... I will commit one macro `PHP_GRPC_FREE_CLASS_OBJECT` again to reduce the follow code:

    php_grpc_zend_object create_wrapped_grpc_timeval(zend_class_entry *class_type
                                                 TSRMLS_DC) {
       PHP_GRPC_ALLOC_CLASS_OBJECT(wrapped_grpc_timeval);
       zend_object_std_init(&intern->std, class_type TSRMLS_CC);
       object_properties_init(&intern->std, class_type);
       PHP_GRPC_FREE_CLASS_OBJECT(wrapped_grpc_timeval, timeval_ce_handlers);
       /*#if PHP_MAJOR_VERSION < 7
         retval.handle = zend_objects_store_put(
           intern, (zend_objects_store_dtor_t)zend_objects_destroy_object,
           free_wrapped_grpc_timeval, NULL TSRMLS_CC);
         retval.handlers = zend_get_std_object_handlers();
         return retval;
      #else
        intern->std.handlers = &timeval_ce_handlers;
        return &intern->std;
        #endif*/
    }

**Comment by thinkerou** [comment]
Hi @stanley-cheung now you should review the PR again and merge it, I will not commit code in the PR commits. Thanks.

**Comment by thinkerou** [comment]
Hi @stanley-cheung we need to add `php7_wrapper.h` file to `src/php/ext/grpc/package.xml` file?

**Comment by stanley-cheung** [comment]
There is no need. Actually we probably can remove that file later. It's not needed

**Comment by thinkerou** [comment]
All changes have committed completely.

File: src/php/ext/grpc/call.c [hunk 2]
```diff
@@ -679,73 +442,37 @@ PHP_METHOD(Call, startBatch) {
...
       recv_md = *grpc_parse_metadata_array(&recv_trailing_metadata);
-      add_property_zval(&recv_status, "metadata", &recv_md);
-      add_property_long(&recv_status, "code", status);
-      add_property_string(&recv_status, "details", status_details);
-      add_property_zval(result, "status", &recv_status);
+      add_property_zval(recv_status, "metadata", &recv_md);
+#endif
+      PHP_GRPC_DELREF(array);
+      add_property_long(recv_status, "code", status);
+      php_grpc_add_property_string(recv_status, "details", status_details, true);
```
Comment (stanley-cheung) [inline src/php/ext/grpc/call.c]: Line over 80 characters

File: src/php/ext/grpc/php7_wrapper.h [hunk 3]
```diff
@@ -49,18 +50,89 @@
...
+  for (zend_hash_internal_pointer_reset(ht); \
+       zend_hash_get_current_data(ht, (void**)&tmp##key) == SUCCESS; \
+       zend_hash_move_forward(ht)) { \
+    key_type = zend_hash_get_current_key_ex(ht, &key, &len##key, &index,\
+                                         0, NULL); \
+    data = *tmp##key;
+
+#define PHP_GRPC_HASH_FOREACH_END() }
+
+static inline int php_grpc_zend_hash_find(HashTable *ht, char *key, int len, void **value) {
```
Comment (stanley-cheung) [inline src/php/ext/grpc/php7_wrapper.h]: Line over 80 characters

**Comment by stanley-cheung** [comment]
LGTM. All relevant tests passed. I will fix the nits in a separate PR. I have a few other minor things I want to fix.

**Comment by thinkerou** [comment]
OK, thank!

**Pull request merged**